### PR TITLE
Graph.ts refactor

### PR DESCRIPTION
**Pull request by Sheyne**

### CODE REVIEW

File: app/components/main/main.component.ts [hunk 1]
```diff
@@ -83,11 +85,11 @@ export class MainComponent implements OnInit, MenuEventListener, REPLDelegate, T
                 this.graphEditor.redraw();
             }
             if (this.pluginService) {
-                if (this.context.graph.pluginManager.kind == "machine-learning.sinap.graph-kind") {
+                if (this.context.graph.graph.plugin.kind == "machine-learning.sinap.graph-kind") {
```
Comment (ilargierdi) [inline app/components/main/main.component.ts]: `context.graph.graph` looks kind of funny.

File: app/components/main/main.component.ts [hunk 2]
```diff
@@ -217,6 +221,16 @@ export class MainComponent implements OnInit, MenuEventListener, REPLDelegate, T
         }
     }
 
+    propertyChanged(event: [PropertiedEntity, keyof PropertiedEntityLists, string, string[]]) {
+        // THIS IS SUPER DIRTY AND CJ SHOULD REALLY HOOK THE CHANGE DETECTOR
+        // TODO: KILL THIS WITH FIRE
```
Comment (ilargierdi) [inline app/components/main/main.component.ts]: Alright, already, jeez.

File: app/models/drawable.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+// File: graph.ts
```
Comment (ilargierdi) [inline app/models/drawable.ts]: Not really important, but this should probably be updated. Also adding a `Contributor` field would be nice, but not necessary.

File: app/components/main/main.component.ts (hunk 2 above)
Comment (Sheyne) [inline app/components/main/main.component.ts]: I'm sorry.

File: app/components/properties-panel/properties-panel.component.ts [hunk 4]
```diff
@@ -3,32 +3,64 @@
...
+
+    @Output() propertyChanged = new EventEmitter();
+
+    // TODO: 
+    // discuss with Daniel
+    // I want to wrap up this functionality into bind.directive
+    // so that we can just do:
+    // <input [snpBind]="selectedEntity" [group]="..." [key]="..." [keyPath]="...">
+    // I've mostly got it, but it needs to somehow link in the forms module
+    // I'll talk more in person.
```
Comment (ilargierdi) [inline app/components/properties-panel/properties-panel.component.ts]: Just bringing this to @RighteousRaichu's attention.

Comment (RighteousRaichu) [inline app/components/properties-panel/properties-panel.component.ts]: Yeah @Sheyne this is something we might want to sit down together and figure out. This area of angular changed considerable from angular 1.x so I'm as lost as you are as to the best way to accomplish this in angular 2.x. 
(Here's a resource explaining some of the changes: https://blog.thoughtram.io/angular/2016/10/13/two-way-data-binding-in-angular-2.html)

**Pull request merged**